### PR TITLE
Add version for maven-compiler-plugin and enable linting output. Fixe…

### PR DESCRIPTION
**Pull request by bmhm**
See comment in: https://github.com/java-native/jssc/issues/34

Sorry for merging that other request early, but I wanted to avoid merging conflicts as I used a new section on both of those PRs.

### CODE REVIEW

**Comment by bmhm** [comment]
Uhm… @tresf why suppress warnings about a `SerialVersionUID`? We should specify one, not supress the warnings. I reverted those changes for now.

File: .travis.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ matrix:
 install:
   - mvn dependency:resolve
 
-script: if [ -z "$PROFILE" ]; then mvn --batch-mode; else mvn -P "$PROFILE" --batch-mode; fi
+script: if [ -z "$PROFILE" ]; then mvn -X -e --batch-mode; else mvn -X -e -P "$PROFILE" --batch-mode; fi
```
Comment (bmhm) [inline .travis.yml]: I think this can now be reverted.

**Comment by tresf** [comment]
> Uhm… @tresf why suppress warnings about a SerialVersionUID

Because the UID created by the JVM is adequate.  We won't be bumping this, it's best to leave to the JVM.

**Comment by bmhm** [comment]
If one of the users wanted to use Serialisation (e.g. by using a cache, RMI or EJB) you would break things.

We should specify our own UID to allow caching and usage in JavaEE.

Yes, generated ones are fine. But if we ship a freshly compiled one, the slightest (even non-breaking) change will cause a new UID. 
I see no reason to implement a weak spot and to discourage use cases for this library, however unlikely.

**Comment by tresf** [comment]
It's a wasted effort because we're not going to change it between API changes.  It's better to ignore it explicitly than claim to support it.  It's not worth arguing though.  If you are going to revert the patch, you should add the value.  Leaving it unconfigured just spams known warnings.

**Comment by tresf** [comment]
Done.  The only thing in the warnings now is the 1.6 stuff.  I'd like to ignore it using `-Xlint:-options` but it suppresses way too much.  I guess we just keep it?

**Comment by bmhm** [comment]
>  The only thing in the warnings now is the 1.6 stuff. I'd like to ignore it using -Xlint:-options but it suppresses way too much. I guess we just keep it?

You mean this output?

> [WARNING] bootstrap class path not set in conjunction with -source 1.6

That will vanish once we compile with `javac` from JDK 9. If we set 9 to the minimum compile JDK, we can use `-release 6` instead of `-source 6 -target 6`. From JDK 9 onwards, all previous Java API signatures are known and inherited when using the `-release` argument on `javac`.

That said, I'd keep it for now in case more shows up. But it wouldn't hurt if we used `-Xlint:-options` and remove that once we start compiling using JDK 9+.

**Comment by tresf** [comment]
>  I'd keep it for now in case more shows up.

Agreed.

> it wouldn't hurt if we used -Xlint:-options and remove that once we start compiling using JDK 9+.

Well, to your previous point, it had suppressed the `serialVersionUID` warnings, so we'll keep it enabled to catch others.

**Pull request merged**